src/components/Navigation/components/Item/tests/Item.test.tsx

```diff
@@ -279,6 +279,73 @@ describe('<Nav.Item />', () => {
         'aria-expanded': true,
       });
     });
+
+    it('invokes the provided onClick handler', () => {
+      const spy = jest.fn();
+      const item = mountWithNavigationProvider(
+        <Item
+          label="some label"
+          url="/admin/orders"
+          subNavigationItems={[
+            {
+              url: '/admin/draft_orders',
+              disabled: false,
+              label: 'draft orders',
+              onClick: spy,
+            },
+          ]}
+        />,
+        {
+          location: '/admin/orders',
+        },
+      );
+
+      item!
+        .find(Secondary)!
+        .find('a')!
+        .trigger('onClick', {
+          preventDefault: jest.fn(),
+          currentTarget: {
+            getAttribute: () => '/admin/draft_orders',
+          },
+        });
+
+      expect(spy).toHaveBeenCalled();
```

Comment:
Might be worth using `toHaveBeenCalledTimes(1)` here. I think nav item is recursive in rendering itself as a sub nav item.